copycat-api/src/main/java/org/apache/kafka/copycat/source/SourceConnector.java

```diff
@@ -0,0 +1,27 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+package org.apache.kafka.copycat.source;
+
+import org.apache.kafka.copycat.connector.Connector;
+
+/**
+ * SourceConnectors implement the connector interface to pull data from another system and send
+ * it to Kafka.
+ */
+public abstract class SourceConnector extends Connector {
```

Comment:
Interesting question. Some sources may need it, but others might auto-generate topic names. For example, a JDBC connector may just construct topic names as something like `<db_name>-<table_name>`. So I'm not sure we want to *require* a TOPIC_CONFIG here.

However, on a related note, we may need to have some interface for getting the topics being published to. This is important if you would want to do something like combine an embedded copycat job with a kstream job -- to run the kstream job you need the set of topics that are being written to, which, as described above, might just be some dynamic set that can't be listed until the connector is running.